### DISCUSSION TITLE
Upgrade from V2 to V3?

### DISCUSSION THREAD

**Discussion by Lunchtime0614** [doudar/SmartSpin2k#477] [General]
I see a new version has been released but I'm struggling to determine if there's any reason to upgrade from V2 to V3?  I see the improvements listed on the readme but to the end user - is there a reason/need to upgrade to the new V3?

From the software perspective it doesn't look like there's a different codebase between the two - they are both using the same code?

**Comment by eMadman** [comment]
V3 enables compatibility with Peloton bikes and opens up the possibility of compatibility with other bikes that use serial for their sensor data.  The only real reason you would have for upgrading is if you're on a Peloton bike and want to get the raw data right out of the sensor cable.  

Both use the same firmware so you're not missing out on anything on the firmware front either!

V2 is suitable for a DIYer looking for a fully hand-built option.  
V3 is more or less ready to rock once you have it in an enclosure and connected to a motor.  The PCBs are much harder to self-source since you really need to order a bulk quantity to get some economies of scale

**Reply by Lunchtime0614** [reply]
Interesting - so I am currently using a Peloton...but I've hacked it all up.  Meaning at this point I've turned it into a dumb spin bike but have assioma power pedals, the SS2K V2 and I use Flux (https://github.com/dvmarinoff/Flux) to which I load ZWO files of publicly available rides (either structured workouts or actual recordings of true rides).

Given that - are you saying the V3 can connect to peloton (I assume via a splitter cable like DFC - https://github.com/intelligenate/dfc) to read the output...I'm presuming that SS2K V3 doesn't control the peloton - correct? As in if the Peloton ride calls for zone 3 it'll auto adjust the resistance as prescribed in the ride (like the very latest peloton bikes do)?  My Peloton bike is the 2nd generation I believe - they only have 3 versions I think.

**Reply by eMadman** [reply]
It's as you assume - you can connect the bike's sensor to the V3.  In your case, if you're not using the tablet, no splitter is needed - just plug directly into the breakout cable.

The SmartSpin will get power and cadence data directly from the bike - meaning you can just leave your assiomas on an outdoors bike.  

The Qdomyos-Zwift app has an integration for SS2K and Peloton.  There's a way to make it work with Peloton guided workouts, but I haven't used it personally.  The integration will also work with your V2 and rely on the reported wattage to adjust resistance as needed.

**Reply by Lunchtime0614** [reply]
Thanks for this - I don't really need the guided workouts since I quit paying the peloton subscription when they increased the price so I can't actually use the peloton tablet anymore because as soon as you do they make you sign up for the all access $50 whatever per month.  Instead you can use any other device, and download the ZWO files - granted this isn't plug and play, but anyone that's buying a SS2K is probably a tinkerer anyway.

I may end up getting the V3 just so I can move the assiomas to another bike - instead of swapping the 2nd set of assiomas to other bikes.

Since I don't have the tablet on the Peloton bike anymore - your saying I can plug the V3 right into the bike without the tablet - What I'm getting at is that there's no way that Peloton could make the V3 obsolete by somehow changing the data feed from the sensors?

**Reply by eMadman** [reply]
The sensor is just a dumb sensor without the tablet - there's no way Peloton can obsolete it for you.

Since you already have a V2, hit me up for a PCB kit instead of a full build.  You can migrate your motor and knob insert over with very little effort.  I can't remember if you have a 3d printer or not.  I'll throw in a printed enclosure and screws for assembly if you need 'em.  The rest of the 3d printed parts are unchanged.

**Reply by doudar** [reply]
I'm not so sure they can't OTA FW the bike sensor, but they for sure can't if you leave it unplugged from the tablet :)

**Reply by Lunchtime0614** [reply]
@eMadman  -  Sweet, no I don't have a 3d printer (and I may just get the whole new version based on what @doudar said about the motor below).  Quick question - does the V3 support ANT+ connections or at least two Bluetooth connections?  One of the reasons I'm using the power pedals on the peloton is that they support ANT+ and bluetooth...so they are connected to the SS2K V2 via Bluetooth and to my Garmin watch via ANT+  So My connection diagram is this:

For the ERG of the ride - - pedals > SS2Kv2 > Flux Trainer app
For auto saving of the workout - - pedals > Garmin Watch > Garmin Connect App>Intervals app

So if I get rid of the pedals and just use the peloton's sensors, to make it as seamless as I have it today I'd need two connections, I think the assiomas can handle two Bluetooth connections at once.

Then my connection diagram would be --- Peloton > SS2K > Garmin and Flux Trainer

**Comment by doudar** [comment]
> The sensor is just a dumb sensor without the tablet - there's no way Peloton can obsolete it for you.
> 
> Since you already have a V2, hit me up for a PCB kit instead of a full build. You can migrate your motor and knob insert over with very little effort. I can't remember if you have a 3d printer or not. I'll throw in a printed enclosure and screws for assembly if you need 'em. The rest of the 3d printed parts are unchanged.



> I see a new version has been released but I'm struggling to determine if there's any reason to upgrade from V2 to V3? I see the improvements listed on the readme but to the end user - is there a reason/need to upgrade to the new V3?
> 
> From the software perspective it doesn't look like there's a different codebase between the two - they are both using the same code?

The code is 99% the same. The firmware detects which hardware version is present, sets the correct variables and enables the serial data communications if you're on R3. 

R3 has a better stepper driver (more powerful and more features) which is able to do torque detection and therefore detect endstops without limit switches, but until someone writes the code for that...it's only more powerful. Aside from Peloton communications, R3 mostly exists just to make it easier to build (and the led's flashing inside the window are pretty swanky.)

**Reply by doudar** [reply]
It will support up to 3 BLE servers and 2 BLE clients (I.e. Zwift + a watch) but there's no ant+.
